File: Apis/Main/Models/NHQ/CadetAchvAprs.cs
// Copyright (C) 2022 Andrew Rioux, Glenn Rioux
// 
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
// 
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
// 
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace UnitPlanner.Apis.Main.Models.NHQ;

[Table("NHQ_CadetAchvAprs")]
public class CadetAchvAprs
{
    public int CadetAchvID { get; set; }

    public int CAPID { get; set; }
    public Member Member { get; set; } = null!;

    [StringLength(3)]
    public string Status { get; set; } = null!;

    public int AprCAPID { get; set; }

    [StringLength(15)]
    public string DspReason { get; set; } = null!;

    public int AwardNo { get; set; }

    public bool JROTCWaiver { get; set; }

    [StringLength(25)]
    public string UsrID { get; set; } = null!;

    public DateTime DateMod { get; set; }

    [StringLength(25)]
    public string FirstUsr { get; set; } = null!;

    public DateTime DateCreated { get; set; }

    public bool PrintedCert { get; set; }
}

File: Services/Capwatch/Capwatch.Service/Models/CadetAchvAprs.fs
namespace UnitPlanner.Services.Capwatch.Service.Models

open System
open System.ComponentModel.DataAnnotations
open System.ComponentModel.DataAnnotations.Schema
open Microsoft.EntityFrameworkCore

[<TableAttribute("NHQ_CadetAchvAprs")>]
type CadetAchvAprs<'a>(CadetAchvID : int, CAPID : int, Member : 'a, Status : string, AprCAPID : int, DspReason : string,
                        AwardNo : int, JROTCWaiver : bool, UsrID : string, DateMod : DateTime, FirstUsr : string,
                        DateCreated : DateTime, PrintedCert : bool) =

    [<Key>]
    member this.CadetAchvID = CadetAchvID

    member this.CAPID = CAPID
    [<ForeignKey("ORGID")>]
    member this.Member = Member

    member this.Status = Status

    member this.AprCAPID = AprCAPID

    member this.DspReason = DspReason

    member this.AwardNo = AwardNo 

    member this.JROTCWaiver = JROTCWaiver

    member this.UsrID = UsrID

    member this.DateMod = DateMod

    member this.FirstUsr = FirstUsr

    member this.DateCreated = DateCreated

    member this.PrintedCert = PrintedCert